.travis.yml

```diff
@@ -0,0 +1,45 @@
+sudo: required
+
+language: python
+python:
+  - 3.8
+
+services:
+  - docker
+
+install:
+  #- docker build --target development --tag $USERNAME/todo-app:$TRAVIS_COMMIT .
+  #- docker build --target production --tag $USERNAME/todo-app:latest .
+  - docker build --target test --tag $USERNAME/todo-app:$TRAVIS_COMMIT .
+  #- docker images
+  
+  #- docker run --env-file ./.env.template -p 5000:80 --mount type=bind,source=/devops-course-starter/src/app/,target=/devops-course-starter/src/app/ --name app todo-app:$TRAVIS_COMMIT
+  #- docker run --env-file ./.env.template -p 5000:80 --mount type=bind,source=/devops-course-starter/src/app/,target=/devops-course-starter/src/app/ --name app todo-app:latest
+  - docker run --env-file ./.env.test -p 5000:80 --mount type=bind,source=/devops-course-starter/src/app/,target=/devops-course-starter/src/app/ --name int-test-app todo-app:$TRAVIS_COMMIT
+
+before_script:
+  - poetry install
+
+script:
+#- docker ps | grep -q app
+  - docker ps | grep -q int-test-app
+  - poetry run pytest
+  - poetry run pytest test_integration_trello.py
+  - poetry run pytest ./test_e2e
+
+after_success:
+  - if [ "$TRAVIS_BRANCH" == "master" ]
```

Comment:
It's worth noting that this could still run against a PR, where travis considers the target branch to be the $TRAVIS_BRANCH. Using the [`deploy` lifecycle stage](https://docs.travis-ci.com/user/deployment/script/) will help with that, and allow you to take advantage of before_deploy for e.g. logging in to the services too

Reply:
As suggested, opted for deploy stage.  Thanx